editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer.tsx

```diff
@@ -220,6 +224,9 @@ export function createComponentRendererComponent(params: {
       highlightBounds: highlightBounds,
       editedText: rerenderUtopiaContext.editedText,
     }
+
+    const buildResult = React.useRef<React.ReactElement | null>(null)
```

Comment:
do we ever read this? maybe we could just simply return the value from the `switch (arbitraryBlockResult.type)` without setting this ref

Reply:
The value of it is returned by the `ComponentRendererComponent` as what should be rendered. I wanted to avoid changing this too much, at least for this PR. I'll make a note to have a look at this separately as it might clean this up a little.